src/compiler/checker.ts

```diff
@@ -16052,9 +16176,13 @@ namespace ts {
                     if (produceDiagnostics) {
                         const symbol = getNodeLinks(node).resolvedSymbol;
                         const typeParameters = symbol.flags & SymbolFlags.TypeAlias ? getSymbolLinks(symbol).typeParameters : (<TypeReference>type).target.localTypeParameters;
-                        checkTypeArgumentConstraints(typeParameters, node.typeArguments);
+                        const minTypeArgumentCount = symbol.flags & SymbolFlags.TypeAlias ? getSymbolLinks(symbol).minTypeArgumentCount : (<TypeReference>type).target.minTypeArgumentCount;
+                        checkTypeArgumentConstraints(typeParameters, node.typeArguments, minTypeArgumentCount);
                     }
                 }
+                if (type.flags & TypeFlags.TypeParameter && !(<TypeParameter>type).isThisType && type.symbol && !isTypeParameterInScope(<TypeParameter>type, node)) {

```

Comment:
Would it be better not to consider type parameters in scope unless they are declared in that particular container type?